gapic/schema/wrappers.py

```diff
@@ -757,17 +757,80 @@ class HttpRule:
     uri: str
     body: Optional[str]
 
-    @property
-    def path_fields(self) -> List[Tuple[str, str]]:
+    def path_fields(self, method: "~.Method") -> List[Tuple[Field, str, str]]:
         """return list of (name, template) tuples extracted from uri."""
-        return [(match.group("name"), match.group("template"))
+        input = method.input
+        return [(input.get_field(*match.group("name").split(".")), match.group("name"), match.group("template"))
                 for match in path_template._VARIABLE_RE.finditer(self.uri)]
 
-    @property
-    def sample_request(self) -> str:
+    def sample_request(self, method: "~.Method") -> str:
```

Comment:
This code needs to be a method of `HttpRule` so that it can touch attributes of the instance; it's really unpleasant to interact with free functions from random modules in the template logic, and almost always better to move necessary logic into a method on the object you're working on.